### ISSUE TITLE
Feature request - Scale per channel

### ISSUE THREAD

**Issue by usbalbin** [hacknus/serial-monitor-rust#92]
When having different values with completely different scales, the smaller ones turns in to flat lines close to zero

It would be useful to have individual scaling per channel like on an oscilloscope. Not shure how that would work with y-ranges etc.

![image](https://github.com/hacknus/serial-monitor-rust/assets/10927717/2d0b8f29-2b5c-42eb-b966-415b48eea7ea)

**Comment by hacknus** [comment]
Indeed a very good idea, this should be implemented! We could have a drop down to select pre-defined gains/scaling factors (..., 0.1, 1, 10, ...) next to the field where you can enter a name for the dataset/curve?

**Comment by usbalbin** [comment]
Perhaps even a "offset" parameter. That, I, believe would be enough to allow scaling and moving around the curves above until they are all evenly spread out and with their respective variations clearly visible.

If one could dream :), there would be two buttons `Autoscale` and `Autoscale side by side`(or similar). Where the first button would scale and move all curves to occupy the hole screen. The second button would do the same but put them side by side so as to avoid any overlapping.

**Comment by hacknus** [comment]
Yes, we would definitely require both a scale and offset. `Autoscale` could also be a possibility and could be easy to implement (not sure though). I think both need to be implemented in `line 333` of `gui.rs`:

```rust
for (i, time) in self.data.time[window..].iter().enumerate() {
    let x = *time as f64 / 1000.0;
    for (graph, data) in graphs.iter_mut().zip(&self.data.dataset) {
        if self.data.time.len() == data.len() {
            if let Some(y) = data.get(i + window) {
                graph.push(PlotPoint { x, y: *y as f64 });   // implement offset and scale here
            }
        }
    }
}
```
